src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -320,17 +320,21 @@ def parse_results(self, results):
         res = []
         for r in results:
             parsed_data = {}
-            for k, v in r[1].items():
-                try:
-                    v = list(i.decode() for i in v)
-                except Exception:
-                    v = list(str(i) for i in v)
-                parsed_data.update({k: v})
-
-            res.append({
-                'dn': r[0],
-                'data': parsed_data
-            })
+            try:
+                for k, v in r[1].items():
+                    try:
+                        v = list(i.decode() for i in v)
+                    except Exception:
+                        v = list(str(i) for i in v)
+                    parsed_data.update({k: v})
+
+                res.append({
+                    'dn': r[0],
+                    'data': parsed_data
+                })
+            except AttributeError:
```

Comment:
Do you think we should perhaps check if `r[1]` exists and is a dict instead ? We can log the other case where that doesn't happen. It just seems clearer to see as to where the issue lies